src/service/routing/core.clj

```diff
@@ -1,6 +1,41 @@
 (ns service.routing.core
-  (:require [compojure.api.sweet :refer :all]
-            [service.routing.spec]))
+  (:require [clojure.spec.gen.alpha :as gen]
+            [ring.util.http-response :refer [ok]]
+            [compojure.api.sweet :refer [context GET api]]
+            [service.routing.spec :as s]
+            [service.routing.directions :as dir]
+            [service.routing.graph.generators :as g]))
+            ;[expound.alpha :as expound]
+
+;(s/valid? (s/and ::radiuses-regex #(= (count coordinates) (count %)))))
+;(s/and ::s/radiuses-regex #(= (count coordinates) (count %))))
+
+(def routes
+  (context "/spec" []
+    :tags ["spec"]
+    :coercion :spec
+
+    (GET "/direction/:coordinates" []
+      :summary "direction with clojure.spec"
+      :path-params [coordinates :- ::s/coordinate-regex]
```

Comment:
@mehdisadeghi I think this is incorrect (although I would need to check). Since you already required the namespace and gave it an alias `s` then you only need to do `:s/coordinate-regex`. `s` gets automatically expanded to the right namespace since it is an alias. `::` is short `:current.namespace/keyword-name` so currently it would expand to something like `:service.routing.core/service.routing.spec/coordinate-regex` which is not what you want I think. 

You can check in the REPL easily by printing the keyword. Again not sure if I am correct ;)